playbooks/roles/ocp-lso/tasks/main.yaml

```diff
@@ -0,0 +1,245 @@
+- name: Create a target namespace
+  kubernetes.core.k8s:
+    state: present
+    definition:
+      apiVersion: v1
+      kind: Namespace
+      metadata:
+        name: "{{ lso_namespace }}"
+
+# Disable all default operator sources
+- name: Disable all default operator sources
+  command: oc patch operatorhub.config.openshift.io/cluster -p='{"spec":{"disableAllDefaultSources":true}}' --type=merge
+  register: patch_operatorhub
+  failed_when: patch_operatorhub.rc != 0  # Fail if the patch operation fails
+
+# Create ICSP & login to podman
+- name: Local Storage Operator deployment
+  block:
+    - name: Setup a catalogsource for installing local storage operator
+      block:
+        - name: Include the global pull-secret update role to extract podman-secret
+          include_role:
+            name: global-secret-update
+
+    - name: Include role to create ImageContentSourcePolicy and CatalogSource
+      include_role:
+        name: set-custom-catalogsource
+      vars:
+        custom_catalogsource_name: "{{ lso_catalogsource_name }}"
+        custom_catalogsource_display_name: "lso-catalog"
+        custom_catalogsource_image: "{{ lso_catalogsource_image }}"
+      when: lso_catalogsource_image != '' and lso_catalogsource_image != None
+
+# Create LocalStorage Operator YAML content (OperatorGroup and Subscription)
+- name: Create operator group for LocalStorage Operator
+  kubernetes.core.k8s:
+    state: present
+    definition:
+      apiVersion: operators.coreos.com/v1
+      kind: OperatorGroup
+      metadata:
+        name: local-operator-group
+        namespace: "{{ lso_namespace }}"
+      spec:
+        targetNamespaces:
+          - "{{ lso_namespace }}"
+
+# Create Subscription for the LocalStorage Operator
+- name: Create LocalStorage Operator Subscription
+  kubernetes.core.k8s:
+    state: present
+    definition:
+      apiVersion: operators.coreos.com/v1alpha1
+      kind: Subscription
+      metadata:
+        name: local-storage-operator
+        namespace: "{{ lso_namespace }}"
+      spec:
+        channel: "{{ lso_channel }}"
+        installPlanApproval: Automatic
+        name: local-storage-operator
+        source: "{{ lso_catalogsource_name }}"
+        sourceNamespace: openshift-marketplace
+
+- name: Wait for CSV to be created
+  pause:
+    seconds: 30 
+
+# Check LocalStorage Operator progress
+- name: Check LocalStorage Operator progress
+  command: oc get ip -n "{{ lso_namespace }}"
+  register: ip_status
+  failed_when:
+    - ip_status.rc != 0  # Fail if the command fails
+    - ip_status.stdout == ""  # If no installation plan is found
+
+# Check the ClusterServiceVersion (CSV) for the LocalStorage Operator
+- name: Check LocalStorage Operator CSV
+  shell: "oc get csv -A | grep local-storage-operator"
+  register: csv_status
+  failed_when: csv_status.rc != 0  # Fail if the command fails
+  when:
+    - lso_namespace is defined  # Ensure lso_namespace is defined
```

Comment:
lso_namespace should be checked while creating the namespace. Also, you can set default namespace.